compiler/tests-jsoo/lib-effects/dune

```
@@ -0,0 +1,19 @@
+(env
+ (_
+  (js_of_ocaml
+   (flags
+    (:standard --enable effects))
+    ;; separate compilation doesn't work when using
+    ;; features such as 'effects' or 'use-js-string'
+    ;; because dune doesn't know that it should compile
+    ;; multiple versions of the dependencies
+   (compilation_mode whole_program))))
```

This seems to work, though `unhandled_unlinked` does not pass since, with separate compilation, the whole standard library is included.
```suggestion
  (js_of_ocaml
   (flags
    (:standard --enable effects))
   (build_runtime_flags (:standard --enable effects)))))
```

---

related to my other question https://github.com/ocsigen/js_of_ocaml/pull/1340#issuecomment-1335261147. Is it ok to mix `--enable effects` and `--disable effects` when doing separate compilation ?

---

There is #943 to track the issue mentioned above

---

Once we disable effect by default, stdlib.cma.js (and other deps) will be compiled without effects support. Testing this, I'm getting lots of errors

---

This restriction will be gone with the next dune 3.7 (see https://github.com/ocaml/dune/pull/6714)